pontoon/contributors/views.py

```diff
@@ -148,14 +134,32 @@ def save_custom_homepage(request):
 def settings(request):
     """View and edit user settings."""
     if request.method == 'POST':
-        form = forms.UserLocalesOrderForm(request.POST, instance=request.user.profile)
-        if form.is_valid():
-            form.save()
+        locales_form = forms.UserLocalesOrderForm(
+            request.POST,
+            instance=request.user.profile,
+        )
+        profile_form = forms.UserProfileForm(
+            request.POST,
+            instance=request.user,
+        )
+        user = get_object_or_404(User, username=request.user.username)
+
+        if locales_form.is_valid() and profile_form.is_valid():
+            locales_form.save()
+            profile_form.save()
+
             messages.success(request, 'Settings saved.')
-            return redirect(request.POST.get('return_url', '/'))
+
+            if user.email != request.user.email:
```

Comment:
Do we need this check? `UserProfileForm` should be invalid if the condition is true, so we never reach this code.

Reply:
If the user didn't change their email, then there's no reason to disconnect them. Unless I'm missing something, this line is needed for that reason, and it does get called. It's possible for the user to edit their name and not their email, in which case the form is valid but the email did not change.

Reply:
You're correct. Sorry for the noise.